src/realm/group.cpp

```diff
@@ -504,6 +504,22 @@ void Transaction::upgrade_file_format(int target_file_format_version)
         }
         remove_table(progress_info->get_key());
     }
+    if (current_file_format_version == 10 && target_file_format_version >= 11 &&
+        get_replication()->get_history_type() == Replication::HistoryType::hist_SyncClient) {
+        // If the file is version 10, then we have to check for missing search index on
+        // string primary key columns. If index is not present, we will try to deduce the
+        // ObjKey from the primary key value by hashing. This will often fail as we use a
+        // different hashing function than was used in core-5.
```

Comment:
....So we'll add the missing index...

otherwise the comment leaves the reader wondering if the problem is still there or is fixed